cypress/integration/dib-travel/personal-settings/profile.spec.ts

```diff
@@ -0,0 +1,221 @@
+    ).select(profileDetails.birthYear);
+    cy.get('.cdk-overlay-container dib-travel-document-dialog dib-input input[name="idNumber"]').type(
+      profileDetails.documentNumber
+    );
+    cy.get('.cdk-overlay-container dib-travel-document-dialog dib-input input[placeholder="Issuing country"]').type(
+      profileDetails.issuingCountry
+    );
+    cy.get('.cdk-overlay-container dib-dialog-wrapper dib-travel-document-dialog dib-searchable-select')
+      .contains(profileDetails.issuingCountry)
+      .click();
+    cy.get(
+      '.cdk-overlay-container dib-travel-document-dialog dib-select-dob[ng-reflect-name="expiryDate"] select.day'
+    ).select(profileDetails.expiryDay);
+    cy.get(
+      '.cdk-overlay-container dib-travel-document-dialog dib-select-dob[ng-reflect-name="expiryDate"] select.month'
+    ).select(profileDetails.expiryMonth);
+    cy.get(
+      '.cdk-overlay-container dib-travel-document-dialog dib-select-dob[ng-reflect-name="expiryDate"] select.year'
+    ).select(profileDetails.expiryYear);
+    cy.get('.cdk-overlay-container dib-travel-document-dialog dib-input input[placeholder="Nationality"]').type(
+      profileDetails.nationality
+    );
+    cy.get('.cdk-overlay-container dib-travel-document-dialog dib-searchable-select div')
+      .contains(profileDetails.nationality)
+      .click();
+    cy.get('.cdk-overlay-container dib-travel-document-dialog ui-checkbox-wrapper').click();
+    cy.get('.cdk-overlay-container dib-travel-document-dialog ui-button').click();
+    cy.get('dib-profile dib-account dib-travel-documents').should('contain', 'PASSPORT');
+    cy.get('dib-profile dib-account dib-travel-documents').should('contain', profileDetails.issuingCountry);
+    cy.get('dib-profile dib-account .travel-documents')
+      .contains(profileDetails.issuingCountry)
+      .next()
+      .contains('Edit')
+      .click({ force: true });
+    cy.get('.cdk-overlay-container dib-travel-document-dialog dib-input input[placeholder="Issuing country"]')
+      .clear()
+      .type(profileDetails.newIssuingCounty);
+    cy.get('.cdk-overlay-container dib-dialog-wrapper dib-travel-document-dialog dib-searchable-select')
+      .contains(profileDetails.newIssuingCounty)
+      .click();
+    cy.get('.cdk-overlay-container dib-travel-document-dialog ui-checkbox-wrapper').click();
+    cy.get('.cdk-overlay-container dib-travel-document-dialog ui-button').click();
+    cy.get('dib-profile dib-account dib-travel-documents').should('contain', 'PASSPORT');
+    cy.get('dib-profile dib-account dib-travel-documents').should('contain', profileDetails.newIssuingCounty);
+    cy.get('dib-profile dib-account .travel-documents')
+      .contains(profileDetails.newIssuingCounty)
+      .next()
+      .contains('delete')
+      .click({ force: true });
+    cy.get('.cdk-overlay-container confirmation-dialog ui-button[type=warning').click();
+  });
+
+  it('should add loyalty program and delete it', () => {
```

Comment:
Some UI/UX loading/rendering delays causing broken test in some cases. Try to investigate more